test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/BatchTests.cs

```diff
@@ -27,11 +40,63 @@ public BatchTests(HttpIntegrationTestFixture fixture)
 
         [Fact]
         [Trait(Traits.Priority, Priority.One)]
-        public async Task WhenSubmittingABatch_GivenAProperBundle_ThenNotSupportedIsReturned()
+        public async Task WhenSubmittingABatch_GivenAProperBundle_ThenSuccessIsReturned()
         {
-            FhirException ex = await Assert.ThrowsAsync<FhirException>(() => Client.PostBundleAsync(Samples.GetDefaultBatch().ToPoco<Bundle>()));
+            Resource requestBundle = Samples.GetDefaultBatch().ToPoco<Bundle>();
 
-            Assert.Equal(HttpStatusCode.MethodNotAllowed, ex.StatusCode);
+            FhirResponse<Bundle> fhirResponse = await Client.PostBundleAsync(requestBundle);
+            Assert.NotNull(fhirResponse);
+            Assert.Equal(HttpStatusCode.OK, fhirResponse.StatusCode);
+            ValidateResourceStatusCode(fhirResponse.Resource);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task WhenSubmittingABatch_GivenAProperBundleWithFewFailedRequests_ThenSuccessIsReturnedAndOperationOutcomesForFailedRequests()
+        {
+            Resource requestBundle = Samples.GetDefaultBatch().ToPoco<Bundle>();
+
+            FhirResponse<Bundle> fhirResponse = await Client.PostBundleAsync(requestBundle);
+            Assert.NotNull(fhirResponse);
+            Assert.Equal(HttpStatusCode.OK, fhirResponse.StatusCode);
+            foreach (var entry in fhirResponse.Resource.Entry)
+            {
+                if (entry.Response.Status.StartsWith("4"))
```

Comment:
We should verify that we got the expected number of exceptions and that they were in the expected index.

Reply:
This method just verifies that operationOutcome is never empty for the exceptions. All HttpStatusCodes(including exceptions) and its index are verified as part of WhenSubmittingABatch_GivenAProperBundle_ThenSuccessIsReturnedForBatchAndExpectedStatusCodesPerRequests method.

Reply:
Instead of only checking for failed operation, can we assert for both success and failures here and validating the operation outcome message for failed cases?

Reply:
Since success and failure status codes for each entry in bundle are verified in another test method. I think it would be redundant if I assert for success status codes here as well for the same bundle.. Thoughts?